### PR TITLE
Update CNI plugins to v1.3.0

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Updating CNI plugins after v1.28.0.
#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
PTAL @kubernetes/release-engineering 
#### Does this PR introduce a user-facing change?

```release-note
Updated CNI plugins to v1.3.0.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by saschagrunert** [comment]
Ran:

```console
kpromo gh --org containernetworking --repo plugins --bucket k8s-artifacts-cni --release-dir release --tags v1.3.0
```

/retest

**Review by haircommander — APPROVED** [review]
/lgtm

File: build/dependencies.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ dependencies:
 
   # CNI plugins
   - name: "cni"
-    version: 1.2.0
+    version: 1.3.0
```
Comment (SergeyKanzhelev) [inline build/dependencies.yaml:21]: need to include this:

https://github.com/kubernetes/kubernetes/blob/b1e35d5616e066c4b0b3f550c33bbf67bb3f876e/hack/local-up-cluster.sh#L55

Comment (saschagrunert) [reply build/dependencies.yaml:21]: Fixed and updated.

**Comment by saschagrunert** [comment]
/skip

This job does not look healthy: https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-local-e2e

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**